src/Microsoft.DotNet.Build.Tasks/CreateAzureContainer.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public sealed class CreateAzureContainer : Task
+    {
+        [Required]
+        public string AccountName { get; set; }
+
+        [Required]
+        public string AccountKey { get; set; }
+
+        [Required]
+        public string ContainerName { get; set; }
+
+        [Output]
+        public string ReadOnlyToken { get; set; }
+
+        [Output]
+        public string StorageUri { get; set; }
+
+        [Output]
+        public string WriteOnlyToken { get; set; }
+
+        public override bool Execute()
+        {
+            Log.LogMessage(MessageImportance.High, "Creating container named '{0}' in storage account {1}.", ContainerName, AccountName);
+
+            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(string.Format("DefaultEndpointsProtocol=https;AccountName={0};AccountKey={1}", AccountName, AccountKey));
+            CloudBlobClient blobClient = storageAccount.CreateCloudBlobClient();
+            CloudBlobContainer newContainer = blobClient.GetContainerReference(ContainerName);
+            newContainer.CreateIfNotExists();
+            StorageUri = newContainer.Uri.ToString();
+
+            // if ReadOnlyToken or WriteOnlyToken != null they should contain the number of days before the token expires
+
+            if (ReadOnlyToken != null)
```

Comment:
Is it actually possible to have a parameter be both input and output?  I don't ever recall seeing something like that in other MSBuild tasks.  Even if it is possible, it might confuse people, since it changes format.  As input it's "string of an integer that is number of days", and output it's "query parameter without the ?".